lib/primitives/address.js

```diff
@@ -386,6 +423,9 @@ class Address {
 
     const [hrp, version, hash] = bech32.decode(data);
 
+    assert(version === 0,
+      'Cannot convert non-program address to bech32m.');
```

Comment:
Hm this error would be thrown if the address IS program but version 1 or higher, so maybe check for -1 only right here, then add a second check that asserts version 0 with the error message:

`'Cannot convert program version > 1 to bech32'`

Also your error message says bech32m! It should just be bech32 right?

Reply:
Yeah, it should be bech32!